src/game/msg_builder.cc

```diff
@@ -71,6 +71,20 @@ ProviderMsgPtr MsgBuilder::CreateNotifyMsgArgs(int seqId, ErrorNumber err,
     return msg;
 }
 
+ProviderMsgPtr MsgBuilder::CreateUserOperationArgs(int seqId, int roomId,
+    int userId, const UserOperation &custom)
+{
+    auto msg = std::make_shared<ProviderMsg>();
+    msg->set_sequenceid(seqId);
+
+    auto args = msg->mutable_useroperationargs();
```

Comment:
I understand that auto could handle &, * properly.
But I still prefer to make the type explicit.  like auto*,  auto&.

it's up to you. if you don't mind, just leave it.